### PR TITLE
MEN-3040: Defined a single-artifact-file workflow

### PR DESCRIPTION
**Pull request by oleorhagen**
This adds a workflow for the 'workflows' microservices which runs a CLI workflow
in order to create the single-file-update Artifact.

Changelog: None

Signed-off-by: Ole Petter <ole.orhagen@northern.tech>

### CODE REVIEW

**Comment by oleorhagen** [comment]
Ticket: https://tracker.mender.io/browse/MEN-3040

**Comment by oleorhagen** [comment]
@tranchitella Is this along the lines of what you had in mind? Or am I completely off the mark here?

File: workflows/single-file-artifact.json [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+            "cli": {
+                "command": [
+                    "create-artifact-worker",
+                    "single-file",
+                    "--artifact-name ${workflow.input.artifact_name}",
+                    "--artifact-id ${workflow.input.artifact_id}",
+                    "--device-type ${workflow.input.device_type}",
+                    "--tenant-id ${workflow.input.tenant_id}",
+                    "--token ${workflow.input.token}",
+                    "--args ${workflow.input.args}",
```
[the remark this answers is not in the document]
Comment (oleorhagen) [reply workflows/single-file-artifact.json]: Alright, thanks. I will continue to fumble my way through this. Don't expect a finished PR anytime soon :p

[the remark this answers is not in the document]
Comment (mchalski) [reply workflows/single-file-artifact.json]: @oleorhagen you may be esp. interested in this pr, it defines the interface:
https://github.com/mendersoftware/create-artifact-worker/pull/3

but it's best to just clone the repo, build the binary and see the current `--help`.

[the remark this answers is not in the document]
Comment (oleorhagen) [reply workflows/single-file-artifact.json]: Yes, I've had a look at it, and as far as I can tell, this workflow covers that interface completely, or am I completely out of the loop here? @mchalski

[the remark this answers is not in the document]
Comment (mchalski) [reply workflows/single-file-artifact.json]: @oleorhagen keep an eye on a last minute change here: https://tracker.mender.io/browse/MEN-3039?focusedCommentId=104157&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-104157

there will be 2 extra args in the cli:
https://github.com/mendersoftware/deployments/pull/386/commits/b271c44e4249f507ae7e65d3f681d39e938d8ede#diff-1b071f23e13699d7f9c09f42692f578cR150

this is in progress now, but you might want to be aware (and just dry-implement those).

[the remark this answers is not in the document]
Comment (oleorhagen) [reply workflows/single-file-artifact.json]: @mchalski better?

File: workflows/single-file-artifact.json [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    "name": "Single-file updates workflow",
+    "description": "Runs a single CLI command -- An invocation of the create_artifact CLI",
+    "version": 1,
+    "tasks": [
+        {
+            "name": "Run create_artifact CLI",
+            "type": "cli",
+            "cli": {
+                "command": [
+                    "create-artifact-worker",
```
Comment (oleorhagen) [inline workflows/single-file-artifact.json]: I see that the `CREATE_ARTIFACT_SERVER` env is required. should it just be an input parameter, and passed onto every single call on the CLI?
Like: `CREATE_ARTIFACT_SERVER=${workflow.input.artifact_server}` you reckon?

Comment (mchalski) [reply workflows/single-file-artifact.json]: @oleorhagen no no - these are env vars set per environment, at container startup.

**Comment by oleorhagen** [comment]
@mchalski great! As far as I know, @tranchitella is testing with this PR as we speak, so I guess I will leave it unmerged.
thx!

**Comment by tranchitella** [comment]
Superseded by https://github.com/mendersoftware/create-artifact-worker/pull/10

**Pull request closed without merging**